bpf/aya-bpf/src/helpers.rs

```diff
@@ -20,6 +40,89 @@ pub unsafe fn bpf_probe_read<T>(src: *const T) -> Result<T, c_long> {
     Ok(v.assume_init())
 }
 
+/// Read bytes stored at the _userspace_ pointer `src` and store them as a `T`.
+///
+/// The size of `T` should be the same as the size of whatever is behind `src`. Typically,
+/// `T` would be a binding such as a struct or primitive that directly maps to the pointee
+/// of `src`.
+///
```

Comment:
The description is valid, but I think it's a bit cryptic. I'd say something like:

```
/// Reads the memory pointed by the _user space_ pointer `src`.
///
/// Returns a bitwise copy of `mem::size_of::<T>()` bytes stored at the user space address `src`. See 
/// `bpf_probe_read_kernel` for  reading kernel space memory.
///
/// # Example
///
/// ...a one liner here...
///
/// # Errors
/// 
/// On failure, returns a negative error code. 
```